src/python/tests/device_api/coap/test_sensor_values.py

```diff
@@ -45,15 +46,58 @@ def _run(coro):
     return asyncio.get_event_loop().run_until_complete(coro)
 
 
-# http://docs.sqlalchemy.org/en/latest/core/dml.html
 class TestSensorValues:
-    def test_render_post(self, mocker):
+    @staticmethod
+    def __setup(mocker):
         mocker.patch('aiopg.sa.create_engine', return_value=CreateEngineContextManager())
+
+    def test_bad_payload1(self, mocker):
+        """Test Bad Input"""
+        self.__setup(mocker)
+        sv = SensorValues()
+        request = MagicMock()
+        dict_payload = {}
+        request.payload = dumps(dict_payload)
+        return_val = _run(sv.render_post(request))
+        assert return_val.code == aiocoap.numbers.codes.Code.BAD_REQUEST
+
+    def test_bad_payload2(self, mocker):
+        """Test Bad Input"""
+        self.__setup(mocker)
+        sv = SensorValues()
+        request = MagicMock()
+        dict_payload = 'hello world'
+        request.payload = dumps(dict_payload)
+        return_val = _run(sv.render_post(request))
+        assert return_val.code == aiocoap.numbers.codes.Code.BAD_REQUEST
+
+    def test_bad_payload3(self, mocker):
+        """Test missing timestamp"""
+        self.__setup(mocker)
+        sv = SensorValues()
+        request = MagicMock()
+        dict_payload = {'asset':'test'}
+        request.payload = dumps(dict_payload)
+        return_val = _run(sv.render_post(request))
+        assert return_val.code == aiocoap.numbers.codes.Code.BAD_REQUEST
+
+    def test_bad_payload4(self, mocker):
+        """Test missing asset"""
+        self.__setup(mocker)
         sv = SensorValues()
         request = MagicMock()
-        dict_payload = {'jack': 4098, 'sape': 4139}
+        dict_payload = {'timestamp':'2017-01-01T00:00:00Z'}
         request.payload = dumps(dict_payload)
         return_val = _run(sv.render_post(request))
-        assert return_val is not None
-        # assert sqlalchemy.Table.insert.mock.assert_called_once_with('?')
-        # assert MagicMockConnection.execute.mock.assert_called_once_with('?')
+        assert return_val.code == aiocoap.numbers.codes.Code.BAD_REQUEST
+
+    def test_good_payload(self, mocker):
+        """Test Good Input"""
+        self.__setup(mocker)
+        sv = SensorValues()
+        request = MagicMock()
+        dict_payload = {'timestamp':'2017-01-01T00:00:00Z', 'asset':'test'}
+        request.payload = dumps(dict_payload)
+        return_val = _run(sv.render_post(request))
+        assert return_val.code == aiocoap.numbers.codes.Code.VALID
+
```

Comment:
@Terris-Dianomic what do you think, about this?

`import aiocoap.numbers.codes.Code as CoAP_CODES`

```
testdata = [
    ({}, CoAP_CODES.BAD_REQUEST),
    ('hello world', CoAP_CODES.BAD_REQUEST),
    ({'asset':'test'}, CoAP_CODES.BAD_REQUEST),
    ({'timestamp':'2017-01-01T00:00:00Z'}, CoAP_CODES.BAD_REQUEST),
    ({'timestamp':'2017-01-01T00:00:00Z', 'asset':'test'}, CoAP_CODES.VALID)
]


@pytest.mark.parametrize("dict_payload, expected", testdata)
def test_payload(self, mocker, dict_payload, expected)
    self.__setup(mocker)
    sv = SensorValues()
    request = MagicMock()
    request.payload = dumps(dict_payload)
    return_val = _run(sv.render_post(request))
    assert return_val.code == expected
```


OR

```
testdata_bad_requests = [
    ({}),
    ('hello world'),
    ({'asset':'test'}),
    ({'timestamp':'2017-01-01T00:00:00Z'})
]
@pytest.mark.parametrize("dict_payload", testdata_bad_requests)
def test_bad_payload(self, mocker, dict_payload)
    self.__setup(mocker)
    sv = SensorValues()
    request = MagicMock()
    request.payload = dumps(dict_payload)
    return_val = _run(sv.render_post(request))
    assert return_val.code == CoAP_CODES.BAD_REQUEST

testdata_good_requests = [
    ({'timestamp':'2017-01-01T00:00:00Z', 'asset':'test'})
]
@pytest.mark.parametrize("dict_payload", testdata_good_requests)
def test_good_payload(self, mocker, dict_payload)
    self.__setup(mocker)
    sv = SensorValues()
    request = MagicMock()
    request.payload = dumps(dict_payload)
    return_val = _run(sv.render_post(request))
    assert return_val.code == CoAP_CODES.VALID
```

https://docs.pytest.org/en/latest/example/parametrize.html